kernel/indtypes.ml

```diff
@@ -730,48 +390,20 @@ let check_positivity ~chkpos kn env_ar_par paramsctxt finite inds =
 (************************************************************************)
 (* Build the inductive packet *)
 
-(* Allowed eliminations *)
-
-let all_sorts = [InProp;InSet;InType]
-let small_sorts = [InProp;InSet]
-let logical_sorts = [InProp]
-
-let allowed_sorts is_smashed s =
-  if not is_smashed 
-  then (** Naturally in the defined sort.
-	   If [s] is Prop, it must be small and unitary.
-	   Unsmashed, predicative Type and Set: all elimination allowed
-	   as well. *)
-      all_sorts
-  else 
-    match Sorts.family s with
-    (* Type: all elimination allowed: above and below *)
-    | InType -> all_sorts
-    (* Smashed Set is necessarily impredicative: forbids large elimination *)
-    | InSet -> small_sorts
-    (* Smashed to Prop, no informative eliminations allowed *)
-    | InProp -> logical_sorts
-    
-(* Previous comment: *)
-(* Unitary/empty Prop: elimination to all sorts are realizable *)
-(* unless the type is large. If it is large, forbids large elimination *)
-(* which otherwise allows simulating the inconsistent system Type:Type. *)
-(* -> this is now handled by is_smashed: *)
-(* - all_sorts in case of small, unitary Prop (not smashed) *)
-(* - logical_sorts in case of large, unitary Prop (smashed) *)
-
-let arity_conclusion = function
-  | RegularArity (_, c, _) -> c
-  | TemplateArity (_, s) -> mkType s
+let repair_arity indices = function
+  | RegularArity ar -> ar.mind_user_arity
+  | TemplateArity ar -> mkArity (indices,Sorts.sort_of_univ ar.template_level)
 
 let fold_inductive_blocks f =
-  Array.fold_left (fun acc (_,_,lc,(arsign,ar)) ->
-    f (Array.fold_left f acc lc) (it_mkProd_or_LetIn (arity_conclusion ar) arsign))
-
-let used_section_variables env inds =
-  let ids = fold_inductive_blocks
-    (fun l c -> Id.Set.union (Environ.global_vars_set env c) l)
-      Id.Set.empty inds in
+  Array.fold_left (fun acc ((arity,lc),(indices,_),_) ->
+    f (Array.fold_left f acc lc) (repair_arity indices arity))
+
+let used_section_variables env params inds =
```

Comment:
Why do we iterate over the params here? Is this a semantic change or just part of the refactoring?

Reply:
Indeed the params are included in the arity/lc so it's pointless. Removed.